boot/freeldr/freeldr/ntldr/winldr.c

```diff
@@ -583,14 +595,17 @@ LoadWindowsCore(IN USHORT OperatingSystemVersion,
              * Parse the port name.
              * Format: /DEBUGPORT=COM[1-9]
              * or: /DEBUGPORT=FILE:\Device\HarddiskX\PartitionY\debug.log
+             * or: /DEBUGPORT=SCREEN
              * or: /DEBUGPORT=FOO
              * If we only have /DEBUGPORT= (i.e. without any port name),
              * defaults it to "COM".
              */
             RtlStringCbCopyA(KdTransportDllName, sizeof(KdTransportDllName), "KD");
-            if (_strnicmp(Option, "COM", 3) == 0 && '0' <= Option[3] && Option[3] <= '9')
+            if ((OptionLength == 4 && _strnicmp(Option, "COM", 3) == 0 && '1' <= Option[3] && Option[3] <= '9') ||
+                (OptionLength > 5 && _strnicmp(Option, "FILE", 4) == 0 && Option[4] == ':') ||
+                (OptionLength == 6 && _strnicmp(Option, "SCREEN", 6) == 0))
```

Comment:
Now, if we happen (or some user happen) to introduce a genuine "kdscreen.dll" (for example) debug transport dll file, you are forbidding it from loading?

Please don't change code you don't fully understand why it is there.